torch/csrc/distributed/c10d/init.cpp

```diff
@@ -700,7 +701,32 @@ They are used in specifying strategies for reduction collectives, e.g.,
           "wait",
           &::c10d::ProcessGroup::Work::wait,
           py::arg("timeout") = kNoTimeout,
-          py::call_guard<py::gil_scoped_release>());
+          py::call_guard<py::gil_scoped_release>())
+      .def(
+          "get_future",
+          [](::c10d::ProcessGroup::Work& work)
+              -> std::shared_ptr<jit::PythonFutureWrapper> {
+            return std::make_shared<jit::PythonFutureWrapper>(work.getFuture());
+          },
+          py::call_guard<py::gil_scoped_release>(),
```

Comment:
`getFuture` seems to be very light-weight, if so, we don't need to release the GIL so that we can avoid additional context switch.

Reply:
makes sense.